src/improvements/template/TransferOwners.sol

```diff
@@ -0,0 +1,137 @@
+/// Some chains may not have a PermissionedWETH or PermissionlessWETH so we handle this accordingly.
+/// ATTENTION: Please use caution when using this template. Transferring ownership is high risk.
+contract TransferOwners is L2TaskBase {
+    using stdToml for string;
+    using LibString for string;
+
+    /// @notice New owner address. This is unaliased.
+    address public newOwner;
+
+    /// @notice PermissionedWETH address.
+    address public permissionedWETH;
+
+    /// @notice PermissionlessWETH address.
+    address public permissionlessWETH;
+
+    /// @notice Stores the chain information after setup.
+    SuperchainAddressRegistry.ChainInfo internal activeChainInfo;
+
+    /// @notice Returns the safe address string identifier.
+    function safeAddressString() public pure override returns (string memory) {
+        return "ProxyAdminOwner";
+    }
+
+    /// @notice Returns the storage write permissions required for this task.
+    function _taskStorageWrites() internal pure virtual override returns (string[] memory) {
+        string[] memory storageWrites = new string[](4);
+        storageWrites[0] = "DisputeGameFactoryProxy";
+        storageWrites[1] = "ProxyAdmin";
+        storageWrites[2] = "PermissionedWETH";
+        storageWrites[3] = "PermissionlessWETH";
+        return storageWrites;
+    }
+
+    /// @notice Sets up the template with the new owner from a TOML file.
+    function _templateSetup(string memory _taskConfigFilePath) internal override {
+        super._templateSetup(_taskConfigFilePath);
+        string memory toml = vm.readFile(_taskConfigFilePath);
+        newOwner = abi.decode(vm.parseToml(toml, ".newOwner"), (address));
+
+        // Only allow one chain to be modified at a time with this template.
+        SuperchainAddressRegistry.ChainInfo[] memory _parsedChains =
+            abi.decode(vm.parseToml(toml, ".l2chains"), (SuperchainAddressRegistry.ChainInfo[]));
+        require(_parsedChains.length == 1, "Must specify exactly one chain id to transfer ownership for");
+        activeChainInfo = _parsedChains[0]; // Store the ChainInfo struct
+    }
+
+    /// @notice Builds the actions for transferring ownership of the DisputeGameFactory, DWETH contracts and ProxyAdmin.
+    function _build() internal override {
+        ProxyAdmin proxyAdmin = ProxyAdmin(superchainAddrRegistry.getAddress("ProxyAdmin", activeChainInfo.chainId));
+        address dgfProxy = superchainAddrRegistry.getAddress("DisputeGameFactoryProxy", activeChainInfo.chainId);
+        permissionedWETH = _getDWETH("PermissionedWETH", activeChainInfo.chainId);
+        permissionlessWETH = _getDWETH("PermissionlessWETH", activeChainInfo.chainId);
+
+        // Transfer ownership of the DisputeGameFactory to the new owner.
+        performOwnershipTransfer(dgfProxy, newOwner);
+
+        // Transfer ownership of the PermissionedWETH to the new owner.
+        if (permissionedWETH != address(0)) {
```

Comment:
This will break after U16 - in U16 DelayedWETH no longer has an owner, it uses the proxy admin owner

Reply:
Makes sense. @mds1 I'm thinking of making this more generic, i.e. if the DelayedWETH contract has an owner function then attempt the owner transfer. If not, then it must be post U16 and skip the owner transfer.

Reply:
that sounds good to me

Reply:
Added fix here: https://github.com/ethereum-optimism/superchain-ops/pull/972/commits/bbb89f7df5312da167d2beb858954c85e8cac3af